File: captainhcg/go/9-palindrome-number.go
func isPalindrome(x int) bool {
    if x < 0{
        return false
    }
    y := x
    rev := 0
    for ; y > 0; {
        a := y % 10
        rev = rev * 10 + a
        y = y / 10
    }
    return rev == x
}

File: bwu/math/9-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        p, res = x, 0
        while p:
            res = res * 10 + p % 10
            p /= 10
        return res == x
